### PR TITLE
Add shared GHA for ruby

### PR DESCRIPTION
**Pull request by girijaasoni**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,11 +3,12 @@ dev_gemfile = File.expand_path("Gemfile.dev.rb", __dir__)
...
-  gem 'rubocop-performance'
-end
+gem 'rake', '~> 13.0'
+gem 'rubocop-rake', '~> 0.6.0'
+
+gem 'gettext', '>= 3.1.3', '< 4.0.0'
 
 group :test do
-  gem 'rake', '~> 10.1.0'
+  gem 'rubocop', '~> 1.57.0'
```
[the remark this answers is not in the document]
Comment (girijaasoni) [reply Gemfile:11]: Thanks Leos, updated!

File: hammer_cli_foreman_kubevirt.gemspec [hunk 2]
```diff
@@ -14,9 +14,6 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib"]
 
   s.add_dependency 'hammer_cli_foreman', '>=0.17.0'
```
Comment (nofaralfasi) [inline hammer_cli_foreman_kubevirt.gemspec:16]:
```suggestion
  s.add_dependency 'hammer_cli_foreman', '~> 3.10'
```

File: .rubocop.yml [hunk 3]
```diff
@@ -1,116 +1,26 @@
-# TODO: remove this file by either moving cops here or fixing code
 require:
   - rubocop-performance
 
-inherit_from:
```
Comment (nofaralfasi) [inline .rubocop.yml:5]: If you drop this, then you should also delete `.rubocop_todo.yml`

File: hammer_cli_foreman_kubevirt.gemspec [hunk 4]
```diff
@@ -14,9 +14,6 @@ Gem::Specification.new do |s|
...
-  s.add_development_dependency 'gettext', '>= 3.1.3', '< 4.0.0'
-  s.add_development_dependency 'rake', '~> 12.3'
-  s.add_development_dependency 'rdoc'
```
Comment (nofaralfasi) [inline hammer_cli_foreman_kubevirt.gemspec:19]: Do you know what this gem is used for? I noticed there's a related task in the `Rakefile`.

File: Gemfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (nofaralfasi) [reply Gemfile:11]: I only see this gem required in `smary_proxy_*` plugins.

File: hammer_cli_foreman_kubevirt.gemspec (hunk 4 above)
Comment (girijaasoni) [reply hammer_cli_foreman_kubevirt.gemspec:19]: I'm not sure what it's used for but I did not remove it, I have added the development dependency in the gemfile

Comment (nofaralfasi) [reply hammer_cli_foreman_kubevirt.gemspec:19]: Where? I don't see it.

Comment (girijaasoni) [reply hammer_cli_foreman_kubevirt.gemspec:19]: [here](https://github.com/theforeman/hammer-cli-foreman-kubevirt/pull/26/commits/5e817e8db9a73095766478abad7f3c8b7557c660#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR8)

Comment (girijaasoni) [reply hammer_cli_foreman_kubevirt.gemspec:19]: Thanks , Updated rdoc in the gemfile, I thought you meant gettext.

**Review by nofaralfasi — APPROVED** [review]

**Review by nofaralfasi — CHANGES_REQUESTED** [review]
Please also update the Rakefile.

**Pull request merged**